### PR TITLE
 Implement a minimal `interruptible` feature

### PR DESCRIPTION
**Pull request by tyrcho**
this is a followup to #50 with the functional interface

### CODE REVIEW

File: src/main/java/com/github/skjolberg/packing/Packager.java [hunk 1]
```diff
@@ -16,6 +18,8 @@
 
 public abstract class Packager {
 
+	static final AtomicBoolean ALWAYS_FALSE = new AtomicBoolean(false);
```
Comment (skjolber) [inline src/main/java/com/github/skjolberg/packing/Packager.java]: So this perhaps not 'on the money' anymore - flavours of BooleanSupplier (always true, deadline, deadline + interrupt) would be perhaps a bit cleaner?

Comment (tyrcho) [reply src/main/java/com/github/skjolberg/packing/Packager.java]: right, I changed it in the last commit

**Comment by tyrcho** [comment]
@skjolber @guizmaii anything else before we can merge this PR ?

**Comment by skjolber** [comment]
@tyrcho looks good to me - are you able to merge yourself now that you're a 'Collaborator'?

**Comment by skjolber** [comment]
I guess doing this 'rotation stream' is fair enough, but it risks wasting time as the packager logic sees the list of containers as a prioritized list. It basically tries to find a better box if that is possible.

**Comment by tyrcho** [comment]
yes, now I can merge, thanks :)

About the rotation I think it's up to you if you want it in the library.

Maybe it's enough to expose it like we did [here](https://github.com/skjolber/3d-bin-container-packing/blob/master/src/main/java/com/github/skjolberg/packing/Container.java#L62) and used in [the test](https://github.com/skjolber/3d-bin-container-packing/blob/master/src/test/java/com/github/skjolberg/packing/PackagerTest.java#L333)

It's actually pretty easy now to pass these 6 rotated Containers to the packager.

**Comment by tyrcho** [comment]
@skjolber what do you think about releasing these new features ?

I can probably push the proper changes in pom.xml and I guess travis will do his magic and publish ... is 1.2.0 alright ?

**Comment by skjolber** [comment]
Release is not done by travis (yet?), running the commands is a manual step. Not really sure how to handle signing the code with my private key via travis.

Merge and I'll release.

**Comment by skjolber** [comment]
Just merge and I'll also fix the version.

**Comment by tyrcho** [comment]
OK, that's clear, and this PR is merged :)

**Comment by skjolber** [comment]
Released! :+1:

**Pull request merged**